src/Iterator/ArrayCacheIterator.php

```diff
@@ -42,7 +46,6 @@ public function __construct(Iterator $iterator)
     /**
      * @return T
      */
-    #[ReturnTypeWillChange]
```

Comment:
should this be removed? This will create a deprecation in PHP 8.1: https://github.com/loophp/collection/pull/225

Reply:
I removed them because I haven't seen them popping in the continuous integration... could you try on your side?

Reply:
I don't have my new laptop set up yet 😅. But in a previous project running on 8.1 they appear as deprecations when running unit tests using [Symfony PHPUnit Bridge](https://symfony.com/doc/current/components/phpunit_bridge.html), which reports deprecations

Reply:
Ok, same here. Let's wait a bit then!

Reply:
This is documented though, and there's no downside to not doing it. See https://www.php.net/manual/en/migration81.incompatible.php#migration81.incompatible.core.type-compatibility-internal.

Every iterator needs to conform to the [Iterator](https://www.php.net/manual/en/class.iterator) interface, which means the return types need to be specified inline on the methods. However, `mixed` as return type is not supported in PHP 7.4, which is why I went with this approach [in this PR](https://github.com/loophp/collection/pull/225)

Reply:
Okay, going to add them everywhere.

Reply:
Done.